source/server/server.h

```diff
@@ -200,6 +200,14 @@ class InstanceImpl : Logger::Loggable<Logger::Id::main>, public Instance {
   void startWorkers();
   void terminate();
 
+  // init_manager_ must come before any member that participates in initialization, since
```

Comment:
The way I read these comments, it feels like they more strongly draw attention to initialization order (top-to-bottom) rather than destruction order (bottom-to-top). I might write something like:

> init_manager_ must appear before any other member that participates in initialization, so it is constructed before they reference it, and destructed only after they can no longer reference it.

... and likewise for secret_manager_.